### PR TITLE
Enable CI on `windows-latest`

### PR DESCRIPTION
**Pull request by marcofavoritobi**
## Proposed changes

This PR makes the test CI workflow to run over windows-latest.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

n/a

### CODE REVIEW

**Comment by marcofavoritobi** [comment]
Tests failing on Windows: https://github.com/bancaditalia/black-it/actions/runs/4451921576/jobs/7819104632?pr=48#step:6:3367

```
FAILED tests/test_calibrator.py::TestCalibrate::test_calibrator_calibrate[2]
FAILED tests/test_calibrator.py::TestCalibrate::test_calibrator_calibrate[1]
FAILED tests/test_examples/test_docker_sir.py::TestDockerSirMainExample::test_run
FAILED black_it/loss_functions/gsl_div.py::black_it.loss_functions.gsl_div.GslDivLoss.discretize
FAILED tests/test_samplers/test_xgboost.py::test_xgboost_2d - assert False
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, windows-latest]
```
Comment (AldoGl) [inline .github/workflows/test.yml:16]: 👍🏼

**Review by AldoGl — APPROVED** [review]
LGTM, thanks.  After this and #47 we should probably advertise that we now support Windows and MacOS on the GitHub page and on the website, but we can do this in another PR

**Comment by marcofavoritobi** [comment]
Thanks @AldoGl !

**Pull request merged**